packages/flutter/lib/src/painting/debug.dart

```diff
@@ -13,6 +12,16 @@ import 'package:flutter/foundation.dart';
 /// This is useful when writing golden file tests (see [matchesGoldenFile]) since
 /// the rendering of shadows is not guaranteed to be pixel-for-pixel identical from
 /// version to version (or even from run to run).
+///
+/// It can be used within tests to enable shadows for a given test for screenshots.
+///
+/// The only way this property works is if it is set to false in a single
```

Comment:
This is not true, there are other ways to use it. You could say something like: In those tests, this is usually set to false at the beginning of a test and back to true before the end of the test case. If it remains true when the test ends, an exception is thrown to avoid state leaking from one test case to another.